Doxyfile

```diff
@@ -1,4 +1,4 @@
-# Doxyfile 1.8.15
+# Doxyfile 1.8.17
```

Comment:
There are other files we should probably re-generate with latest doxygen as well (1.8.18):
https://github.com/Azure/azure-sdk-for-c/blob/3094430d27c9bbf5461082d5070f3c91969822c6/eng/docs/api/assets/footer.html#L1
https://github.com/Azure/azure-sdk-for-c/blob/1048107b737833101994c6954151fcafd80da8df/eng/docs/api/Doxyfile.template#L1
https://github.com/Azure/azure-sdk-for-c/blob/901de0d372000178ded9a0c6dc5943c15288b318/eng/docs/api/assets/header.html#L1
https://github.com/Azure/azure-sdk-for-c/blob/901de0d372000178ded9a0c6dc5943c15288b318/eng/docs/api/assets/style.css#L1

Reply:
https://github.com/Azure/azure-sdk-for-c/issues/745